R/importance_sampling.R

```diff
@@ -0,0 +1,219 @@
+#' A parent class for different importance sampling methods.
+#' @inheritParams psis
+#' @template is_method
+#' @keywords internal
+importance_sampling <- function(log_ratios, method, ...) UseMethod("importance_sampling")
+
+#' @noRd
+#' @keywords internal
+#' @description
+#' Currently implemented importance sampling methods
+assert_is_method_is_implemented <- function(x){
```

Comment:
using `is` for two different things in the function name is perhaps suboptimal. It is ok for me just perhaps confusing to people reading the code for the first time.

Reply:
Indeed. Ill make this more clear.